internal/powerplatform/services/environment/api_environment.go

```diff
@@ -420,7 +423,10 @@ func (client *EnvironmentClient) UpdateEnvironment(ctx context.Context, environm
 		return nil, err
 	}
 
-	client.Api.Sleep(10 * time.Second)
+	err = client.Api.SleepWithContext(ctx, 10*time.Second)
```

Comment:
All these hard-coded time durations would probably be better managed in a variable somewhere.

Reply:
Agree ... will create a separate issue / PR to track this since it's outside the scope of the billing policy bug I was trying to fix.